MainModule/Server/Commands/Moderators.lua

```diff
@@ -4928,6 +4928,90 @@ return function(Vargs, env)
 			end
 		};
 
+		CreateLeaderstats = {
+			Prefix = Settings.Prefix;
+			Commands = {"createstat", "cstat"};
+			Args = {"name"};
+			Description = "Creates a new leaderstat";
+			AdminLevel = "Moderators";
+			Function = function(plr: Player, args: {string})
+
+				local statName = assert(args[1], "Missing stat name")
+
+				for _, v in service.GetPlayers(plr, args[1]) do
+					local leaderstats = plr:FindFirstChild("leaderstats")
+
+					if leaderstats and (leaderstats:IsA("Folder") or leaderstats:IsA("StringValue")) then
+						service.New("IntValue", {
+							Parent = leaderstats,
+							Name = args[1],
+							Value = 0
+						})
+					else
+						service.New("Folder", {
+							Parent = plr,
+							Name = "leaderstats"
+						})
+
+						service.New("IntValue", {
+							Parent = leaderstats,
+							Name = args[1],
+							Value = 0
+						})
+					end
+				end
+			end
+		};
+
+		DeleteLeaderstats = {
+			Prefix = Settings.Prefix;
+			Commands = {"deletestat", "dstat"};
+			Args = {"name"};
+			Description = "Deletes a leaderstat";
+			AdminLevel = "Moderators";
+			Function = function(plr: Player, args: {string})
+				local statName = assert(args[1], "Missing the stat name!")
+
+				for _, v in service.GetPlayers(plr, args[1]) do
+					local children = plr:GetChildren()
+					local leaderstats = plr.leaderstats -- This checks if there is a leaderstats folder/string value [Don't change it unless you know what you are doing]
```

Comment:
Another issue. This won't actually check if leaderstats exists... it's just going to error and break the entire loop/command if it *doesn't* exist already. That's aside from the issue that it's currently running this on the player who ran the command with a number of loops equal to the number of players that it should have been ran on...

Reply:
I just fixed this